stakingModel/contracts/ContributionCalculation.sol

```diff
@@ -69,19 +51,19 @@ contract ContributionCalculation is DSMath, SchemeGuard {
     * `sellContributionRatio` percent contribution
     * @return (contributionAmount) the contribution amount for sell
     */
-    function calculateContribution(ERC20 token, uint256 gdAmount)
+    function calculateContribution(address _marketMaker, address _reserve, ERC20 _token, uint256 _gdAmount)
```

Comment:
should be of type Reserve and type MarketMaker
also lets add the address of the user as input. ie "address _contributer"

Reply:
updated bb6e5869d2f98e6a1debe2e1c86708469a5d2270